src/internal/components/QrCode/QrCodeSvg.tsx

```diff
@@ -0,0 +1,183 @@
+  logoBorderRadius?: number;
+  quietZone?: number;
+  quietZoneBorderRadius?: number;
+  ecl?: 'L' | 'M' | 'Q' | 'H';
+  gradientType?: 'radial' | 'linear';
+};
+
+export function QrCodeSvg({
+  value,
+  size = QR_CODE_SIZE,
+  backgroundColor = '#ffffff',
+  logo,
+  logoSize = QR_LOGO_SIZE,
+  logoBackgroundColor = QR_LOGO_BACKGROUND_COLOR,
+  logoMargin = 5,
+  logoBorderRadius = QR_LOGO_RADIUS,
+  quietZone = 12,
+  quietZoneBorderRadius = 10,
+  ecl = 'Q',
+  gradientType = 'radial',
+}: QRCodeSVGProps) {
+  const gradientRadius = size * 0.55;
+  const gradientCenterPoint = size / 2;
+
+  const { config } = useOnchainKit();
+  const ockTheme = (config?.appearance?.theme ?? 'default') as
+    | 'default'
+    | 'base'
+    | 'cyberpunk'
+    | 'hacker';
+
+  const isRadialGradient = gradientType === 'radial';
+  const fillColor = isRadialGradient ? 'url(#radialGrad)' : '#000000';
+  const bgColor = isRadialGradient
+    ? backgroundColor
+    : `url(#${gradientType}Grad)`;
+
+  const linearGradientColor = ockThemeToLinearGradientColorMap[
+    ockTheme
+  ] as keyof typeof linearGradientStops;
+  const linearColors = [
+    linearGradientStops[linearGradientColor].startColor,
+    linearGradientStops[linearGradientColor].endColor,
+  ];
+
+  const presetGradientForColor =
+    presetGradients[
+      ockThemeToRadiamGradientColorMap[ockTheme] as keyof typeof presetGradients
+    ];
+
+  const matrix = useMatrix(value, ecl);
+  const corners = useCorners(size, matrix.length, bgColor, fillColor);
+  const { x: x1, y: y1 } = GRADIENT_START_COORDINATES;
+  const { x: x2, y: y2 } = GRADIENT_END_COORDINATES;
+
+  const viewBox = useMemo(() => {
+    return [
+      -quietZone,
+      -quietZone,
+      size + quietZone * 2,
+      size + quietZone * 2,
+    ].join(' ');
+  }, [quietZone, size]);
+
+  const svgLogo = useLogo({
+    size,
+    logo,
+    logoSize,
+    logoBackgroundColor,
+    logoMargin,
+    logoBorderRadius,
+  });
+
+  const path = useDotsPath({
+    matrix,
+    size,
+    logoSize,
+    logoMargin,
+    logoBorderRadius,
+    hasLogo: !!logo,
+  });
+
+  if (!path || !value) {
+    return null;
+  }
+
+  return (
+    <svg viewBox={viewBox} width={size} height={size}>
+      <title>QR Code</title>
+      <defs>
+        {isRadialGradient ? (
+          <radialGradient
+            id="radialGrad"
+            data-testid="radialGrad"
+            rx={gradientRadius}
+            ry={gradientRadius}
+            cx={gradientCenterPoint}
+            cy={gradientCenterPoint}
+            gradientUnits="userSpaceOnUse"
+          >
+            {presetGradientForColor.map(([gradientColor, offset]) => (
+              <stop
+                key={`${gradientColor}${offset}`}
+                offset={offset}
+                stopColor={gradientColor}
+                stopOpacity={1}
+              />
+            ))}
+          </radialGradient>
+        ) : (
+          <linearGradient
+            id="linearGrad"
```

Comment:
concat with [useId](https://react.dev/reference/react/useId) to ensure no conflicts